lib/Parse/ParseExpr.cpp

```diff
@@ -568,9 +559,12 @@ ParserResult<Expr> Parser::parseExprKeyPath() {
 
   // FIXME: diagnostics
   ParserResult<Expr> rootResult, pathResult;
+  Optional<ParserStatus> parseStatus;
```

Comment:
You can simply use `ParserStatus parseStatus` here. The default is success. What I was talking is that `ParserResult` cannot be success without the value. But `ParserStatus` can have simple "success" value.